libparsec/crates/client/tests/unit/workspace/folder_transactions.rs

```diff
@@ -476,9 +477,45 @@ async fn ops_outbound_sync(ops: &WorkspaceOps) {
     }
 }
 
+fn rename_entry_by_id(
+    ops: Arc<WorkspaceOps>,
+    parent_id: VlobID,
+    _base_path: FsPath,
+    src: EntryName,
+    dst: EntryName,
+) -> BoxFuture<'static, ()> {
+    Box::pin(async move {
+        ops.rename_entry_by_id(parent_id, src, dst, MoveEntryMode::CanReplace)
+            .await
+            .unwrap();
+    })
+}
+
+fn move_entry_same_parent(
+    ops: Arc<WorkspaceOps>,
+    _parent_id: VlobID,
+    base_path: FsPath,
+    src: EntryName,
+    dst: EntryName,
+) -> BoxFuture<'static, ()> {
+    Box::pin(async move {
+        let src = base_path.join(src);
+        let dst = base_path.join(dst);
+        ops.move_entry(src, dst, MoveEntryMode::CanReplace)
+            .await
+            .unwrap();
+    })
+}
+
+type RenameFlavor =
+    fn(Arc<WorkspaceOps>, VlobID, FsPath, EntryName, EntryName) -> BoxFuture<'static, ()>;
+
 /// Test that renaming a file to match the prevent sync pattern make the file disappear from the parent in the remote
 #[parsec_test(testbed = "coolorg", with_server)]
+#[case::rename_entry_by_id(rename_entry_by_id)]
```

Comment:
replace this by `#[values("rename", "move")] kind: &str`

then you can simplify do:

```rust
match kind {
  "rename" => {
       ops.rename_entry_by_id(parent_id, src, dst, MoveEntryMode::CanReplace)
            .await
            .unwrap();
  }
  "move" => {
    ops.move_entry(src, dst, MoveEntryMode::CanReplace)
            .await
            .unwrap();
  }
 unknown => panic!("Unknown kind: {}", unknown),
}
```